### DISCUSSION TITLE
How to access current user role in the Plugin `initialize()` method?

### DISCUSSION THREAD

**Discussion by lc-at** [kanboard/kanboard#5175] [Q&A] [answered]
I am currently tweaking a Kanboard theme plugin. I want the theme to be applied only when the user is not an admin or not logged in.

I have tried this, but did not work.

```php
public function initialize()
{
    if ($this->userSession->isAdmin()) return;
    $this->template->setTemplateOverride(...);
    ... stripped ...
}
```

Probably because the session is not active when the plugins were loaded. If so, what is the correct way to satisfy my intention (initialize theme only when user is not admin)?

**Comment by creecros** [comment, marked as answer]
maybe this will help you:
https://github.com/creecros/Customizer/blob/8200b2fb4091ac445333a83d5d8bf9914065d758/Plugin.php#L131-L139

it sets the theme to the users theme. I put it in `onStartup()`. 
using `$this->userSession->getId();` which will not work in the Plugin.php, which is why i threw it in a model and then call it.

i think i have noticed it does not work perfectly, its been a long time since i have played around with it, but if my memory serves me, once in a blue moon it takes a second navigation to pick up the session.

**Reply by lc-at** [reply]
I just found a neat little workaround that works (at least) for me.

```php
// Plugin.php, initialize()
session_start([
    'read_and_close'  => true,
]);
if ($this->userSession->isAdmin()) return;
```

**UPDATE**: I guess accessing `$this->userSession` freezes the `userSession` (this error popped: `Cannot override frozen service "userSession"."`), so instead I just use `session_get` directly

```php
if (session_exists('user') && session_get('user')['role'] === 'app-admin')
    return;
```

**Please note that** the above solution is not guaranteed to work across different versions and is just a hacky workaround that is enough to satisfy my need.

I think the way that you suggested may work and is actually the better way, but I don't think I need it at the moment. Thanks! Really appreciate your response.

**Reply by creecros** [reply]
Who knows, I like learning new things, so thanks for that.
